lib/internal/modules/esm/resolve.js

```diff
@@ -897,26 +897,21 @@ function moduleResolve(specifier, base, conditions, preserveSymlinks) {
   // Ok since relative URLs cannot parse as URLs.
   let resolved;
   if (shouldBeTreatedAsRelativeOrAbsolutePath(specifier)) {
-    try {
-      resolved = new URL(specifier, base);
-    } catch (cause) {
-      const error = new ERR_UNSUPPORTED_RESOLVE_REQUEST(specifier, base);
-      setOwnProperty(error, 'cause', cause);
-      throw error;
+    resolved = URL.parse(specifier, base);
+
+    if (resolved == null) {
+      throw new ERR_UNSUPPORTED_RESOLVE_REQUEST(specifier, base);
```

Comment:
I'm not sure about this one, it seems useful to have the error as cause (also if we're throwing an error anyway, I don't think we need to worry too much about performance)

Reply:
Throwing an error, catching it, and later throwing it again has a really big performance impact. I'm not quite sure about the un-happy path probability of this code...

Reply:
I don't think "passing invalid URL to `import()`" is a use-case we want to support.

> Throwing an error, catching it, and later throwing it again has a really big performance impact

You say "really big", but I'm unconvinced we could even measure it.